### DISCUSSION TITLE
for anyone suffering slow traffic when play music

### DISCUSSION THREAD

**Discussion by spd789562** [Androz2091/discord-player#873] [General]
Recently normal youtube-dl will face some traffic problems, for python users can use `yt-dlp` instead
but you can use extractor for discord-player which is [downloader](https://github.com/DevSnowflake/discord-player-downloader) by the following command
```shell
YOUTUBE_DL_HOST=https://api.github.com/repos/yt-dlp/yt-dlp/releases?per_page=1 npm install @discord-player/downloader
```
or add dependence in package.json and just
```shell
YOUTUBE_DL_HOST=https://api.github.com/repos/yt-dlp/yt-dlp/releases?per_page=1 npm install 
```
notice: `use cross-env in windows`

then inside package `youtube-dl-exec` will download yt-dlp instead of ytdl when it install

and just follow [downloader](https://github.com/DevSnowflake/discord-player-downloader#with-discord-player) instruction
```js
const downloader = require("@discord-player/downloader").Downloader;

player.use("YOUTUBE_DL", downloader);
```

it also resolves my bot play longer than 10 minutes video need a longer timeout, otherwise , it'll just stop after few seconds

**Comment by Majoramari** [comment]
didn't work got an error

**Reply by spd789562** [reply]
what did you got?

**Reply by Majoramari** [reply]
I did try again on another system, it works I tested it on a 20-min video 👍🏻
